### PR TITLE
Do not Initialize language workers for dotnet functions

### LINKED ISSUE #2946: Don't start Grpc server when not needed
My function app only contains a single C# function. However, in the logs I see lots of Grpc failures (I was doing a load test):

```
System.IO.IOException : Failed to bind port "127.0.0.1:0"
   at Grpc.Core.Server.CheckPortsBoundSuccessfully()
   at Grpc.Core.Server.Start()
   at Microsoft.Azure.WebJobs.Script.Grpc.GrpcServer.StartAsync() at C:\azure-webjobs-sdk-script\src\WebJobs.Script.Grpc\Server\GrpcServer.cs : 36
   at Microsoft.Azure.WebJobs.Script.ScriptHost.InitializeWorkers() at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Host\ScriptHost.cs : 716
   at Microsoft.Azure.WebJobs.Script.ScriptHost.Initialize() at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Host\ScriptHost.cs : 305
   at Microsoft.Azure.WebJobs.Script.ScriptHostManager.RunAndBlock(CancellationToken cancellationToken) at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Host\ScriptHostManager.cs : 196
```

We should only be starting up this server if it is needed (i.e. other languages are being used).

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #2946

### CODE REVIEW

File: src/WebJobs.Script/Utility.cs [hunk 1]
```diff
@@ -447,6 +448,20 @@ internal static bool IsSingleLanguage(IEnumerable<FunctionMetadata> functions, s
...
+        internal static bool ShouldInitiliazeLanguageWorkers(IEnumerable<FunctionMetadata> functions, string language)
+        {
+            if (!string.IsNullOrEmpty(language) && language.Equals(LanguageWorkerConstants.DotNetLanguageWorkerName, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+            if (functions != null && functions.Any())
+            {
+                List<string> dotNetLanguages = new List<string>() { DotNetScriptTypes.CSharp, DotNetScriptTypes.FSharp, DotNetScriptTypes.DotNetAssembly };
+                return functions.Where(f => !dotNetLanguages.Contains(f.Language, StringComparer.OrdinalIgnoreCase)).Any();
```
Comment (fabiocav) [inline src/WebJobs.Script/Utility.cs]: Remove the where clause and just pass the lambda into the `Any` call. Should actually be able to simplify this by using `All` instead (avoid the negation)

File: src/WebJobs.Script/Description/DotNet/DotNetScriptTypes.cs [hunk 2]
```diff
@@ -5,9 +5,9 @@ namespace Microsoft.Azure.WebJobs.Script.Description
 {
     public class DotNetScriptTypes
     {
-        public const string CSharp = "CSharp";
-        public const string FSharp = "FSharp";
-        public const string DotNetAssembly = "DotNetAssembly";
-        public const string RawDotNetAssembly = "RawDotNetAssembly";
+        public const string CSharp = "csharp";
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/DotNetScriptTypes.cs]: With all comparisons being case insensitive, is the casing change here needed? We log this data and this change may impact queries/groupings

File: src/WebJobs.Script/Utility.cs [hunk 3]
```diff
@@ -447,6 +448,20 @@ internal static bool IsSingleLanguage(IEnumerable<FunctionMetadata> functions, s
             return true;
         }
 
+        internal static bool ShouldInitiliazeLanguageWorkers(IEnumerable<FunctionMetadata> functions, string language)
```
Comment (fabiocav) [inline src/WebJobs.Script/Utility.cs]: Same feedback as the `_language` in `ScritpHost`

File: src/WebJobs.Script/Description/DotNet/DotNetScriptTypes.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Description/DotNet/DotNetScriptTypes.cs]: Reverted this change. Changed to lower case to be consistent with "language" coming from worker.config.json files.

**Pull request merged**